apps/fz_http/lib/fz_http/rules.ex

```diff
@@ -19,14 +20,27 @@ defmodule FzHttp.Rules do
 
   def as_settings do
     Repo.all(
-      from r in Rule, select: %{destination: r.destination, user_id: r.user_id, action: r.action}
+      from r in Rule,
+        select: %{
+          destination: r.destination,
+          user_id: r.user_id,
+          action: r.action,
+          port_range: r.port_range,
+          port_type: r.port_type
+        }
     )
     |> Enum.map(&setting_projection/1)
     |> MapSet.new()
   end
 
   def setting_projection(rule) do
-    %{destination: decode(rule.destination), user_id: rule.user_id, action: rule.action}
+    %{
+      destination: decode(rule.destination),
+      user_id: rule.user_id,
+      action: rule.action,
+      port_range: rule.port_range,
+      port_type: rule.port_type
+    }
```

Comment:
Maybe could be shortened?

```suggestion
rule
|> Map.take([:user_id, :action, :port_range, :port_type])
|> Map.merge(%{destination: decode(rule.destination)})
```